core/kernel.rbs

```diff
@@ -2647,7 +2644,7 @@ module Kernel : BasicObject
   #     Object.new.nil?   #=> false
   #     nil.nil?          #=> true
   #
-  def nil?: () -> bool
+  def nil?: () -> false
```

Comment:
Should this be `false` or `bool`? `NilClass` overrides it to `true`

Reply:
It should be `bool` to maintain polymorphism. See https://github.com/ruby/rbs/issues/1822#issuecomment-2120990437

Reply:
It's fine to have `false` here.
I don't think maintaining polymorphism is a good reason here, because Ruby allows (and many classes are already) breaking *Liskov substitution principle* by `undef` and incompatible override.

Reply:
By polymorphism, my concern is that for `x.nil?` to typecheck correctly, it must be explicitly `Object?` to trigger typecheckers’ special handing. If we type `x` as `Object`, `x.nil?` would typecheck to only `false` even when `x` becomes `nil`.

Reply:
Oh... my bad. Will open another PR to fix the problem.